test/output_test.dart

```diff
@@ -162,5 +162,16 @@ selector[href*=\"{\"] { /* please don't move me */ }
 
 @rule3;"""));
     });
+
+    test("loud comments in mixin", () {
```

Comment:
This case raises a couple other possible edge cases:

* What if the loud comment comes after the style rule but still isn't contained within it, for example when the style rule is also in a mixin?
* What if the loud comment is in a totally different file from the mixin?

I'd probably solve these by writing a `bool spanContainsLocation(SourceSpan span, SourceLocation location)` helper and checking that in the outer conditional.